### PR TITLE
37 implement validation

### LINKED ISSUE #37: [Implement] - Validation
Implementing validation in our application enhances data integrity, security, user experience, and business logic enforcement. It ensures consistent data quality, reduces error handling overhead, aligns with industry standards, integrates seamlessly with frameworks, and improves code maintainability.

### PR DESCRIPTION
**Pull request by marcusju23**
Closes #37 .

### CODE REVIEW

**Comment by marcusju23** [comment]
I also noticed that I tagged someone by mistake in commit [3666cc0](https://github.com/Lucia560/Jakarta_labb/pull/45/commits/3666cc000a865ab6861492b0b22d83c21b1e4b88). Anyone that knows how to resolve this?

**Comment by BarbieQkiller** [comment]
> I also noticed that I tagged someone by mistake in commit [3666cc0](https://github.com/Lucia560/Jakarta_labb/pull/45/commits/3666cc000a865ab6861492b0b22d83c21b1e4b88). Anyone that knows how to resolve this?

Rollback perhaps?

**Review by Lucia560 — DISMISSED** [review]
Well done:)

**Comment by BarbieQkiller** [comment]
@marcusju23 I just wanted to let you know that I am going through everything now :)

Before submitting the review I just wanted to say, yay amazing job!

File: src/main/java/org/example/jakarta_labb/validate/CustomReleaseYearValidator.java [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+
+public class CustomReleaseYearValidator implements ConstraintValidator<ReleaseYear, LocalDate> {
+
+    @Override
+    public void initialize(ReleaseYear constraintAnnotation) {
+        ConstraintValidator.super.initialize(constraintAnnotation);
+    }
+
+    @Override
+    public boolean isValid(LocalDate releaseYear, ConstraintValidatorContext context) {
```
Comment (BarbieQkiller) [inline src/main/java/org/example/jakarta_labb/validate/CustomReleaseYearValidator.java:17]: Should this be an int? In the MovieDto the type is int.

File: src/test/java/org/example/jakarta_labb/dto/MovieDtoTest.java [hunk 2]
```diff
@@ -1,4 +1,4 @@
-package org.example.jakarta_labb;
+package org.example.jakarta_labb.dto;
```
Comment (BarbieQkiller) [inline src/test/java/org/example/jakarta_labb/dto/MovieDtoTest.java:1]: Nice job with the structure reorganization. 👍

File: src/main/java/org/example/jakarta_labb/validate/ReleaseYear.java [hunk 3]
```diff
@@ -0,0 +1,47 @@
+package org.example.jakarta_labb.validate;
```
Comment (BarbieQkiller) [inline src/main/java/org/example/jakarta_labb/validate/ReleaseYear.java:1]: This annotation is not used should it be removed?

**Review by BarbieQkiller — COMMENTED** [review]
This is really great and valuable work. Very well done! 😺

**Review by Lucia560 — APPROVED** [review]
Looks good :)

**Comment by BarbieQkiller** [comment]
> Looks good :) I did not see in your updated pom the maven.surefire.pluggin. I think we still need that:)

When I looked through the code the surefire plugin was there ☺️

File: src/main/java/org/example/jakarta_labb/validate/CustomReleaseYearValidator.java (hunk 1 above)
Comment (marcusju23) [reply src/main/java/org/example/jakarta_labb/validate/CustomReleaseYearValidator.java:17]: Yes you are correct, I was playing around with the LocalDate to try to make it more dynamic. I corrected it now! 😄

File: pom.xml [hunk 4]
```diff
@@ -2,73 +2,90 @@
...
-      </dependency>
+    </dependency>
     <dependency>
       <groupId>org.glassfish.jersey.test-framework.providers</groupId>
       <artifactId>jersey-test-framework-provider-jetty</artifactId>
       <version>3.1.5</version>
       <scope>test</scope>
     </dependency>
-    </dependencies>
+    <dependency>
```
[the remark this answers is not in the document]
Comment (marcusju23) [reply pom.xml:52]: Should be fine for now, but good thing to keep in mind! 👍

File: src/main/java/org/example/jakarta_labb/validate/ReleaseYear.java (hunk 3 above)
Comment (marcusju23) [reply src/main/java/org/example/jakarta_labb/validate/ReleaseYear.java:1]: I would say lets keep it for now.

**Review by BarbieQkiller — APPROVED** [review]

**Comment by marcusju23** [comment]
> ## [Quality Gate Failed](https://sonarcloud.io/dashboard?id=Lucia560_Jakarta_labb&pullRequest=45) **Quality Gate failed**
> 
> Failed conditions  [7.0% Duplication on New Code](https://sonarcloud.io/component_measures?id=Lucia560_Jakarta_labb&pullRequest=45&metric=new_duplicated_lines_density&view=list) (required ≤ 3%)
> 
> [See analysis details on SonarCloud](https://sonarcloud.io/dashboard?id=Lucia560_Jakarta_labb&pullRequest=45)

I would say let's ignore the dupe complaint for now 👍

**Review by Lucia560 — APPROVED** [review]
Nice.)

**Review by TrungMa1 — APPROVED** [review]
Wow. Thats alot of work you've made. There's atleast no issues that my little knowledge can spot 👍

**Pull request merged**